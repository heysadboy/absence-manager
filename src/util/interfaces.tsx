export interface IAbsence {
    admitterId: number | null;
    admitterNote: string | null,
    confirmedAt: string | null,
    createdAt: string | null,
    crewId: number,
    endDate: string | null,
    id: number,
    memberNote: string | null,
    rejectedAt: string | null,
    startDate: string | null,
    type: string | null,
    userId: number
}

export interface IMember {
    crewId: number,
    id: number,
    image: string,
    name: string,
    userId: number
}

export interface IMemberList {
    [id: number]: IMember
}