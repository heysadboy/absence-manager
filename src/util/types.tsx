export type TPageIndex = {
    start: number;
    end: number;
    total: number
}

export type TDateRange = {
    start: string,
    end: string
}