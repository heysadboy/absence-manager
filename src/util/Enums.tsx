export enum EAbsenceStatus {
    requested = 'Requested',
    confirmed = 'Confirmed',
    rejected = 'Rejected'
}

export enum EPageStatus {
    loading = 'Loading',
    error = 'Error',
    success = 'Success',
    empty = 'Empty'
}

export enum EAbsenceType {
    sickness = 'Sickness',
    vacation = 'Vacation',
    all = 'All'
}
