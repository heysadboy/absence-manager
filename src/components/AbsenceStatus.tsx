import { FC } from 'react';
import { IAbsence } from '../util/Interfaces';
import { EAbsenceStatus } from '../util/Enums';

interface IAbsenceStatusProps {
    absence: IAbsence
}

const AbsenceStatus: FC<IAbsenceStatusProps> = ({ absence }) => {
    if (absence.confirmedAt != null) {
        return <div className="ui mini green label">{EAbsenceStatus.confirmed}</div>;
    }
    else if (absence.rejectedAt != null) {
        return <div className="ui mini red label">{EAbsenceStatus.rejected}</div>;
    }
    return <div className="ui mini blue label">{EAbsenceStatus.requested}</div>;
}

export default AbsenceStatus;