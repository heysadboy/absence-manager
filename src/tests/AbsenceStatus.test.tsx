import renderer from 'react-test-renderer';
import  AbsenceStatus from '../components/AbsenceStatus';
import { IAbsence } from '../util/Interfaces';


test('AbsenceStatus.tsx snapshot testing', () => {

  const absence: IAbsence = {
    admitterId: 1,
    admitterNote: "admitter note",
    confirmedAt: "2020-12-12",
    createdAt: "2020-12-12",
    crewId: 1,
    endDate: "2020-12-12",
    id: 1,
    memberNote: "member note",
    rejectedAt: "2020-12-12",
    startDate: "2020-12-12",
    type: "vacation",
    userId: 1
  }

  const component = renderer.create(
    <AbsenceStatus absence={absence}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});