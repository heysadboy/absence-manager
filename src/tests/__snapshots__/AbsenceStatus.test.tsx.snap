// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbsenceStatus.tsx snapshot testing 1`] = `
<div
  className="ui mini green label"
>
  Confirmed
</div>
`;
